src/transformers/integrations.py

```diff
@@ -1299,6 +1305,99 @@ def on_train_end(self, args, state, control, **kwargs):
             self.tracker.stop()
 
 
+class ClearMLCallback(TrainerCallback):
+    """
+    A [`TrainerCallback`] that sends the logs to [ClearML](https://clear.ml/).
+
+    Environment:
+            CLEARML_PROJECT (`str`, *optional*, defaults to "HuggingFace Transformers"):
+                ClearML project name. If not set deafault project name will be assigned.
+            CLEARML_TASK (`str`, *optional* defaults to `"Trainer"`):
+                ClearML task name. If not set default task name will be assigned.
+    """
+
+    def __init__(self):
+        has_clearml = is_clearml_available()
+        if not has_clearml:
+            raise RuntimeError("ClearMLCallback requires 'clearml' to be installed. Run `pip install clearml`.")
+        if has_clearml:
+            import clearml
+            self._clearml = clearml
+        else:
+            self._clearml = None
+
+        self._initialized = False
+        self._clearml_task = None
+
+    def setup(self, args, state, model, tokenizer, **kwargs):
+        if self._clearml is None:
+            return
+        if state.is_world_process_zero:
+            logger.info(
+                'Automatic ClearML logging enabled.'
+            )
+            if self._clearml_task is None:
+                self._clearml_task = self._clearml.Task.init(project_name=os.getenv("CLEARML_PROJECT", "HuggingFace Transformers"),task_name=os.getenv("CLEARML_TASK", "Trainer"), auto_connect_frameworks={'tensorboard': False, 'pytorch': False})
+                self._initialized = True
+                logger.info(
+                    'ClearML Task has been initialized.'
+                )
+
+            self._clearml_task.connect(args, "Args")
+            if hasattr(model, "config") and model.config is not None:
+                self._clearml_task.connect(model.config, "Model Configuration")
+
+
+    def on_train_begin(self, args, state, control, model=None, tokenizer=None, **kwargs):
+        if self._clearml is None:
+            return
+        if state.is_hyper_param_search:
```

Comment:
For this pull request, we previously decided not to add hyperparameter search, as ClearML has already integrated hyperparameter searching libraries which have been included in Hugging Transformer. So, for hyperparameter search for every new run, we are basically creating a new ClearML Task, rather than using ClearML for hyperparameter search. W&B was doing to same in this case also. So, to reduce the complicacy of this pull request we decided not to add hyper-parameter search this time.